pandas/core/window/common.py

```diff
@@ -52,7 +52,8 @@ def __init__(self, obj, *args, **kwargs):
         kwargs.pop("parent", None)
         groupby = kwargs.pop("groupby", None)
         if groupby is None:
-            groupby, obj = obj, obj.obj
+            groupby, obj = obj, obj._obj_with_exclusions
```

Comment:
Hmm yea I find this a little strange; do you see what groupby code is actaully getting executed? I think that we may need to replace a call to `obj` there with `obj_with_exclusions` but I wouldn't store the attributes this way

Reply:
> do you see what groupby code is actaully getting executed?

It's in the `._apply` method off this same class:
```
    def _apply(
        self,
        func: Callable,
        center: bool,
        require_min_periods: int = 0,
        floor: int = 1,
        is_weighted: bool = False,
        name: Optional[str] = None,
        use_numba_cache: bool = False,
        **kwargs,
    ):
        """
        Dispatch to apply; we are stripping all of the _apply kwargs and
        performing the original function call on the grouped object.
        """
        kwargs.pop("floor", None)

        # TODO: can we de-duplicate with _dispatch?
        def f(x, name=name, *args):
            x = self._shallow_copy(x)

            if isinstance(name, str):
                return getattr(x, name)(*args, **kwargs)

            return x.apply(name, *args, **kwargs)
        breakpoint()
        return self._groupby.apply(f)
```

>  but I wouldn't store the attributes this way

Yeah, seems strange to me too, I'll think about a better way to do this

Reply:
Also, like this is wrong, as we have
```
(Pdb) g.obj
    A   B
0   1   0
1   1   1
2   1   2
3   1   3
4   1   4
5   1   5
6   1   6
7   1   7
8   1   8
9   1   9
10  1  10
11  1  11
12  1  12
13  1  13
14  1  14
15  1  15
16  1  16
17  1  17
18  1  18
19  1  19
20  2  20
21  2  21
22  2  22
23  2  23
24  2  24
25  2  25
26  2  26
27  2  27
28  2  28
29  2  29
30  2  30
31  2  31
32  3  32
33  3  33
34  3  34
35  3  35
36  3  36
37  3  37
38  3  38
39  3  39
(Pdb) g.rolling(4)
RollingGroupby [window=4,center=False,axis=0]
(Pdb) g.obj
     B
0    0
1    1
2    2
3    3
4    4
5    5
6    6
7    7
8    8
9    9
10  10
11  11
12  12
13  13
14  14
15  15
16  16
17  17
18  18
19  19
20  20
21  21
22  22
23  23
24  24
25  25
26  26
27  27
28  28
29  29
30  30
31  31
32  32
33  33
34  34
35  35
36  36
37  37
38  38
39  39
(Pdb) 
```

I've made this mistake before so I should know better